### PR TITLE
[1LP][RFR] test_tagvis_combination, test_clone fixes

### PR DESCRIPTION
**Pull request by rlbabyuk**
Purpose or Intent
=================
Updated the way we get vms, and templates, for a test.
{{pytest: -vvv -k 'test_tagvis_tag_datacenter_combination or test_vm_clone' --long-running --use-provider complete}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 9b39ecfe19189de53a5d1aaf7dc1c143701c78be:

`cfme/common/provider_views.py`:
- :bangbang: Line 283:1: **E302** *expected 2 blank lines, found 1*

`cfme/infrastructure/provider/__init__.py`:
- :red_circle: Line 336:1: **F811** *redefinition of unused 'ProviderTemplates' from line 327*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by rlbabyuk** [comment]
tests for 5.9 failed as we have an issue with a partial match for the dropdown, and also a bz that role cannot be added.

**Comment by quarckster** [comment]
@rlbabyuk please add `pytest.skip` decorator with BZ blocker for respectful tests.

**Comment by rlbabyuk** [comment]
@quarckster, bz blocker is added

File: cfme/tests/infrastructure/test_infra_tag_filters_combination.py [hunk 1]
```diff
@@ -33,19 +34,21 @@ def testing_vis_object(request, a_provider, appliance):
...
-
-    if not test_items:
-        pytest.skip('No content found for test!')
-
-    if collection_name in ['templates', 'vms']:
-        item_type = a_provider.data['provisioning']['catalog_item_type'].lower()
-        for resource in collection_rest:
-            if resource.vendor == item_type:
-                return param_class(name=resource.name, provider=a_provider)
+    if not param_class:
```
Comment (psav) [inline cfme/tests/infrastructure/test_infra_tag_filters_combination.py]: **REQUIRED**: Just a question here, why don't we require the destination to be passed in as well with an option for it to be None, which will default to all. This saves the special cases of vms/clusters.

Comment (rlbabyuk) [reply cfme/tests/infrastructure/test_infra_tag_filters_combination.py]: Updated

**Pull request merged**